view/frontend/web/js/product/options.js

```diff
@@ -120,23 +120,15 @@ define([
             if(productType === 'configurable'){
 
                 var selectedOptions = {};
-                var validSuperAttribute = '';
                 $('[name^="super_attribute"]').each(function() {
                     var attributeId = $(this).attr('name').match(/\[(\d+)\]/)[1];
                     selectedOptions[attributeId] = $(this).val();
-                    validSuperAttribute = validSuperAttribute + Boolean(selectedOptions[attributeId]);
                 });
-                validSuperAttribute =  Boolean(validSuperAttribute && validSuperAttribute.search("false") === -1);
-
-                document.customOptions = document.getSkuForOptions(selectedOptions);
-                document.resetPriceAndSku(true);
-
-            }else{
-
-                document.customOptions = document.getSkuForOptions(selectedOptions);
-                document.resetPriceAndSku(true);
 
             }
+
+            document.customOptions = document.getSkuForOptions(selectedOptions);
+            document.resetPriceAndSku(true);
```

Comment:
should this always being returning true here, since that's meaning the item is configurable? Since theres a check before this to only do things to configurable items, it infers that there are also non configurable items that hit this code

Reply:
Yes because simple products can have custom options too.